gaj23/bigbirds
# Gaj/refactor/proptypes

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. Include screenshots as needed.

This includes the addition of propTypes where needed. 
Issue closed #18 

### Problems Encountered & Resulting Solutions  
> State any problems encountered & include screenshots as needed. 
> Describe your code changes in detail for reviewers.

During the application of propTypes, it became apparent that how the 'checked' boolean value was being initially set to `null` was not the best option. Thus, changes were made based on that feedback so that a bird's `checked` value starts of as `false` to ensure that if something other then a boolean value occurs for that variable, then an error will be able to provide better description. 

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

See `Bird` and `Sighting` for main changes. 

### How should this be tested?
> What steps need to be taken to confirm functionality? 

Please start your local sever or go to the host site and open the console. No proptypes errors should be visible. 

### Steps To Take
> Any questions and possible steps moving forward.

As this refactoring began, I've considered ways to also factor out `useLocation` of `Router`. While it was beneficial and interesting to learn about, my inability to check the values that are being passed between reveal the potential weakness of its usage. 

#### Checklist:
